internal/omap/orderedmap_test.go

```diff
@@ -0,0 +1,211 @@
+package omap_test
+
+import (
+	"testing"
+
+	"github.com/shoenig/test"
+	"github.com/shoenig/test/must"
+
+	"github.com/tarantool/go-config/internal/omap"
+	"github.com/tarantool/go-config/internal/testutil"
+)
+
+func TestOrderedMap_Set_Get(t *testing.T) {
+	t.Parallel()
+
+	m := omap.New[string, int]()
+	m.Set("a", 1)
+	m.Set("b", 2)
+	m.Set("c", 3)
+
+	val, ok := m.Get("a")
+	must.True(t, ok)
+	test.Eq(t, 1, val)
+
+	val, ok = m.Get("b")
+	must.True(t, ok)
+	test.Eq(t, 2, val)
+
+	val, ok = m.Get("c")
+	must.True(t, ok)
+	test.Eq(t, 3, val)
+
+	_, ok = m.Get("missing")
+	test.False(t, ok)
+}
```

Comment:
It looks like two or three tests actually:
```suggestion
// single+multiple could be merged:
func TestOrderedMap_Set_Get_single(t *testing.T) {
	t.Parallel()

	m := omap.New[string, int]()
	m.Set("a", 1)

	val, ok := m.Get("a")
	must.True(t, ok)
	test.Eq(t, 1, val)
}

func TestOrderedMap_Set_Get_multiple(t *testing.T) {
	t.Parallel()

	m := omap.New[string, int]()
	m.Set("a", 1)
	m := omap.New[string, int]()
	m.Set("b", 2)

	val, ok := m.Get("a")
	must.True(t, ok)
	test.Eq(t, 1, val)

	val, ok := m.Get("b")
	must.True(t, ok)
	test.Eq(t, 2, val)
}

func TestOrderedMap_Set_Get_missing(t *testing.T) {
	t.Parallel()

	m := omap.New[string, int]()
	m.Set("a", 1)

	_, ok = m.Get("missing")
	test.False(t, ok)
}
```